src/index.js

```diff
@@ -135,8 +137,9 @@ const MicroModal = (() => {
 
     const triggers = document.querySelectorAll('[data-micromodal-trigger]')
     const triggerMap = generateTriggerMap(triggers)
+    const hasLogging = options.debugMode || false
 
-    if (validateArgs(triggers, triggerMap)) {
+    if (validateArgs(triggers, triggerMap, hasLogging)) {
```

Comment:
Let's not pass `hasLogging` to `validateArgs`. This function job is only to validate the markup and throw errors, it should not be concerned with whether debugging is enabled or not. In this case we are still calling the function even if we don't want to execute it.

Reply:
I think a better approach would be to use multiple if conditions

```
    if (options.debugMode === true && validateArgs(triggers, triggerMap) === false) return

    for (var key in triggerMap) {
        ...
    }
```

- Now only if the `debugMode` is set to `true` the `validateArgs` function will fire
- When nothing is specified, `options.debugMode` will evaluate to `undefined` therefore `options.debugMode === true` will return false and the `validateArgs` method will never fire.
- If `validateArgs` fires and the finds errors it will log the errors and then stop execution.

Makes sense?

Reply:
Thank you for such detailed explanation. Made complete sense.
I have incorporated suggestions.